### PR TITLE
Improved readString() for File

### PR DESCRIPTION
**Pull request by apicquot**
readString() method for the class File is inherited from the class Stream that requires a timeout of 1 second at the end of the file, hence a very poor performance. Stream implementation also uses multiple memory resizing that could be anticipated based on the size of the file.

The proposed implementation solves these two issues.

### CODE REVIEW

**Comment by apicquot** [comment]
may want to use a define rather than sizeof for the size of the block to read ?

File: cores/esp8266/FS.cpp [hunk 1]
```diff
@@ -121,6 +121,22 @@ const char* File::name() const {
     return _p->name();
 }
 
+String File::readString()
+{
+    String ret;
+    ret.reserve(size() - position());
+    char temp[256];
+    int countRead = readBytes(temp, sizeof(temp));
+    while (countRead > 0)
```
Comment (devyte) [inline cores/esp8266/FS.cpp]: You're forgetting to read the last chunk when it has less than sizeof(temp). Eg.: consider a file with size 257bytes, this code will only read the first 256, and the last one will be dropped.
You have to calculate the delta remaining.

**Comment by devyte** [comment]
The function sizeof is compile-time  code, so it gets optimized to a single value. Your use is correct. The number 256 shoukd be in a macro, though.

**Comment by devyte** [comment]
ping!

**Comment by apicquot** [comment]
I don't understand, the code works for me for files<256 or files>= 256: I read and push blocks while countRead>0.

The test countRead<sizeof(temp) is just to place a string terminator for the last block.

**Comment by devyte** [comment]
Apologies, my previous review comment was messed up. It's corrected now.

**Comment by apicquot** [comment]
You are absolutely right @devyte : String::concat(char*) needs a string terminator to work. I fixed it.

string::concat(char*, int length) may have been more elegant but it is a protected method

File: cores/esp8266/FS.cpp (hunk 1 above)
Comment (apicquot) [reply cores/esp8266/FS.cpp]: I fixed it, thank you @devyte

**Pull request merged**